### ISSUE TITLE
Support Swin Transformer Model

### ISSUE THREAD

**Issue by tanzhenyu** [keras-team/keras-cv#671]
Related to https://github.com/keras-team/keras-cv/issues/668, this is another transformer based model which we'd like to be supported. For simplicity, focus on only classification task, having in mind that future detection tasks might rely on it.

For contributors, please follow the [contribution guide](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md).

And specifically, on the [API Design Guide](https://github.com/keras-team/keras-cv/blob/master/.github/API_DESIGN.md), we'd like to see a brief description of how the model would look like before implementation, preferrably conforming to the tf.keras.application style.

**Comment by ariG23498** [comment]
I would like to work on this!

**Comment by ariG23498** [comment]
Hey @tanzhenyu

I got around reading the contriution guideline and the API design document. I was unable to find a guide to contribute a model inside of `keras_cv`.

My main doubts:
- Do we need to port the weights of the model (here SwinT in PyTorch)? If yes (which I hope should be the case), how would I do that?
- Do we also need to validate the weight porting process by validating it wrt the task (classification here) on a benchmark dataset?
- How do we check the correctness of the model?

**Comment by bhack** [comment]
> Hey @tanzhenyu
> 
> I got around reading the contriution guideline and the API design document. I was unable to find a guide to contribute a model inside of `keras_cv`.
> 
> My main doubts:
> 
> * Do we need to port the weights of the model (here SwinT in PyTorch)? If yes (which I hope should be the case), how would I do that?
> * Do we also need to validate the weight porting process by validating it wrt the task (classification here) on a benchmark dataset?
> * How do we check the correctness of the model?

It would be useful to have something like:
https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md#contributing-to-keras-applications

**Comment by bhack** [comment]
Also do we have a duplicate ticket with https://github.com/keras-team/keras-cv/issues/636 ?

**Comment by tanzhenyu** [comment]
> Hey @tanzhenyu
> 
> I got around reading the contriution guideline and the API design document. I was unable to find a guide to contribute a model inside of `keras_cv`.
> 
> My main doubts:
> 
> * Do we need to port the weights of the model (here SwinT in PyTorch)? If yes (which I hope should be the case), how would I do that?
> * Do we also need to validate the weight porting process by validating it wrt the task (classification here) on a benchmark dataset?
> * How do we check the correctness of the model?

Yes we do expect porting the weights and verifying the results against the original paper. For the guide, @bhack 's reference is most relevant.

**Comment by ayulockin** [comment]
@ariG23498 will you mind if I join you on this PR?

**Comment by ariG23498** [comment]
I have no issues collaborating with you on this one!

**Comment by bhack** [comment]
> Yes we do expect porting the weights and verifying the results against the original paper. For the guide, @bhack 's reference is most relevant.

We have @sayakpaul https://github.com/sayakpaul/swin-transformers-tf for this

Then we need to build the training phase.

**Comment by DavidLandup0** [comment]
Any progress on this @ayulockin @ariG23498? 
If you'd like to drop it, I can port Sayak's implementation and move onto MaxViT

**Comment by DavidLandup0** [comment]
To do transformer-based models we'll need new layers (reusable) and blocks (might be reusable). KerasCV is currently missing blocks to rapidly build transformer-based models. Might be worth taking a detour to work on those components first

**Comment by tanmaypandey7** [comment]
Hi! Are you guys @ayulockin and @ariG23498 still working on this? I would like to help you guys if it's alright.

**Comment by IMvision12** [comment]
Is Anyone working on this??

**Comment by sachinprasadhs** [comment]
Thanks for reporting the issue! We have consolidated the development of KerasCV into the new KerasHub package, which supports image, text, and multi-modal models. Please read https://github.com/keras-team/keras-hub/issues/1831. KerasHub will support all the core functionality of KerasCV.

KerasHub can be installed with !pip install -U keras-hub. Documentation and guides are available at [keras.io/keras_hub](https://keras.io/keras_hub/).

With our focus shifted to KerasHub, we are not planning any further development or releases in KerasCV. If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).
